pkg/symbol/elfutils/debuginfofile.go

```diff
@@ -130,9 +156,40 @@ func (f *debugInfoFile) SourceLines(addr uint64) ([]profile.LocationLine, error)
 		})
 	}
 
+	name, ok := tr.Entry.Val(dwarf.AttrName).(string)
+	if !ok {
+		name = ""
+	}
+
+	// address correspond line must in last position
+	file, line := findLineInfo(f.lineEntries[cu.Offset], addr)
+	lines = append(lines, profile.LocationLine{
+		Line: line,
+		Function: f.demangler.Demangle(&pb.Function{
+			Name:     name,
+			Filename: file,
+		}),
+	})
+
+	moveLinesForwardOneStep(lines)
```

Comment:
Could you explain why we do this?

Reply:
> 看起来不错，感谢您修复它。
> 
> 是否可以使用小型二进制文件添加更多测试？

here is an idea:
i will get a dwarf file from our production machine and correspond go pprof file.

test:
range each line in go pprof file.
    pass pc to parca and get line.
    compare the parca result with pprof result(which is we expected).

Reply:
> 你能解释一下我们为什么这样做吗？

exemple, a program:
```
0 package main
1
2 func main() {
3      Top()
4 }
5 func Top() {
6       // pc
7  }
```
use dwarf to tran pc.
 the function top is inline subroutine, it would be main.Top main.go 3, (the line number is where the top  be called as inlined).
And the pc's dwarf.LineEntry would be main.main main.go 6. (actual line number)
so the result would be [{main.Top main.go 3}, {main.main main.go 6}]

but the pprof format expected  [{main.Top main.go 6}, {main.main main.go 3}].
it want function line number not where it be called, but itself.

so we move forword the locations's line number one step.

Reply:
We can compile the same program with inlining disabled and enabled and check the binaries into git and use it for testing. To prevent any future regressions? What do you think?

Reply:
> We can compile the same program with inlining disabled and enabled and check the binaries into git and use it for testing. To prevent any future regressions? What do you think?

I find there already have binaries in parca/pkg/symbolizer/testdata.

so  i will find an address in these binaries which  have cross complication inline function,
and add this address to test, is this enough?

And i should use addr2line to tran pc when runtime.OS is linux, make sure debuginfo's output is equal with addr2line output.

Reply:
> And i should use addr2line to tran pc when runtime.OS is linux, make sure debuginfo's output is equal with addr2line output.

This is my ultimate goal for this tests. So it would be amazing.